orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/persistence/ExecutionRepository.java

```diff
@@ -53,41 +52,50 @@
 
   void updateStatus(ExecutionType type, @Nonnull String id, @Nonnull ExecutionStatus status);
 
-  @Nonnull Execution retrieve(@Nonnull ExecutionType type, @Nonnull String id) throws ExecutionNotFoundException;
+  @Nonnull
+  Execution retrieve(@Nonnull ExecutionType type, @Nonnull String id) throws ExecutionNotFoundException;
 
   void delete(@Nonnull ExecutionType type, @Nonnull String id);
 
-  @Nonnull Observable<Execution> retrieve(ExecutionType type);
+  @Nonnull
+  Observable<Execution> retrieve(ExecutionType type);
 
-  @Nonnull Observable<Execution> retrievePipelinesForApplication(@Nonnull String application);
+  @Nonnull
+  Observable<Execution> retrievePipelinesForApplication(@Nonnull String application);
 
-  @Nonnull Observable<Execution> retrievePipelinesForPipelineConfigId(
-    @Nonnull String pipelineConfigId, @Nonnull ExecutionCriteria criteria);
+  @Nonnull
+  Observable<Execution> retrievePipelinesForPipelineConfigId(@Nonnull String pipelineConfigId,
+                                                             @Nonnull ExecutionCriteria criteria);
 
-  @Nonnull Observable<Execution> retrieveOrchestrationsForApplication(
-    @Nonnull String application, @Nonnull ExecutionCriteria criteria);
+  @Nonnull
+  Observable<Execution> retrieveOrchestrationsForApplication(@Nonnull String application,
+                                                             @Nonnull ExecutionCriteria criteria);
 
-  @Nonnull Execution retrieveOrchestrationForCorrelationId(
-    @Nonnull String correlationId) throws ExecutionNotFoundException;
+  @Nonnull
+  Execution retrieveOrchestrationForCorrelationId(@Nonnull String correlationId) throws ExecutionNotFoundException;
 
   @Nonnull
   List<Execution> retrieveBufferedExecutions();
 
-  final class ExecutionCriteria {
+  final class DefaultExecutionCriteria implements ExecutionCriteria {
```

Comment:
I don't understand why this needs to be an interface. Would a different persistence strategy do anything different?

Reply:
Ya, I don't think it needs to be either.

Reply:
I had initially started down the path of exposing a different `ExecutionCriteria` for the polling agents, which had criteria that only the polling agent interface methods would use. Didn't end up needing that, but figured I'd keep it in the PR in case anyone saw usefulness long-term.

Will remove all that.